libs/langchain-community/src/embeddings/jina/index.ts

```diff
@@ -0,0 +1,204 @@
+import { Embeddings, type EmbeddingsParams } from "@langchain/core/embeddings";
+import { chunkArray } from "@langchain/core/utils/chunk_array";
+import { getEnvironmentVariable } from "@langchain/core/utils/env";
+
+export interface JinaEmbeddingsParams extends EmbeddingsParams {
+  /** Model name to use */
+  modelName:
+    | "jina-clip-v2"
+    | "jina-embeddings-v3"
+    | "jina-colbert-v2"
+    | "jina-clip-v1"
+    | "jina-colbert-v1-en"
+    | "jina-embeddings-v2-base-es"
+    | "jina-embeddings-v2-base-code"
+    | "jina-embeddings-v2-base-de"
+    | "jina-embeddings-v2-base-zh"
+    | "jina-embeddings-v2-base-en"
+    | string;
+
+  baseUrl?: string;
+
+  /**
+   * Timeout to use when making requests to Jina.
+   */
+  timeout?: number;
+
+  /**
+   * The maximum number of documents to embed in a single request.
+   */
+  batchSize?: number;
+
+  /**
+   * Whether to strip new lines from the input text.
+   */
+  stripNewLines?: boolean;
+
+  /**
+   * The dimensions of the embedding.
+   */
+  dimensions?: number;
+
+  /**
+   * Scales the embedding so its Euclidean (L2) norm becomes 1, preserving direction. Useful when downstream involves dot-product, classification, visualization..
+   */
+  normalized?: boolean;
+}
+
+type JinaMultiModelInput =
+  | {
+      text: string;
+      image?: never;
+    }
+  | {
+      image: string;
+      text?: never;
+    };
+
+export type JinaEmbeddingsInput = string | JinaMultiModelInput;
+
+interface EmbeddingCreateParams {
+  model: JinaEmbeddingsParams["modelName"];
+
+  /**
+   * input can be strings or JinaMultiModelInputs,if you want embed image,you should use JinaMultiModelInputs
+   */
+  input: JinaEmbeddingsInput[];
+  dimensions: number;
+  task: "retrieval.query" | "retrieval.passage";
+  normalized?: boolean;
+}
+
+interface EmbeddingResponse {
+  model: string;
+  object: string;
+  usage: {
+    total_tokens: number;
+    prompt_tokens: number;
+  };
+  data: {
+    object: string;
+    index: number;
+    embedding: number[];
+  }[];
+}
+
+interface EmbeddingErrorResponse {
+  detail: string;
+}
+
+export class JinaEmbeddings extends Embeddings implements JinaEmbeddingsParams {
+  modelName: JinaEmbeddingsParams["modelName"] = "jina-clip-v2";
+
+  batchSize = 24;
+
+  baseUrl = "https://api.jina.ai/v1/embeddings";
+
+  stripNewLines = true;
+
+  dimensions = 1024;
+
+  apiKey: string;
+
+  normalized = true;
+
+  constructor(
+    fields?: Partial<JinaEmbeddingsParams> & {
+      apiKey?: string;
+    }
+  ) {
+    const fieldsWithDefaults = { maxConcurrency: 2, ...fields };
+    super(fieldsWithDefaults);
+
+    const apiKey =
+      fieldsWithDefaults?.apiKey ||
+      getEnvironmentVariable("JINA_API_KEY") ||
+      getEnvironmentVariable("JINA_AUTH_TOKEN");
+
+    if (!apiKey) throw new Error("Jina API key not found");
+
+    this.apiKey = apiKey;
+
+    this.modelName = fieldsWithDefaults?.modelName ?? this.modelName;
```

Comment:
We usually standardize around `model`, might have missed this on initial review

Might recommend deprecating this field and renaming